win32ss/user/ntuser/main.c

```diff
@@ -815,6 +815,12 @@ ExitThreadCallback(PETHREAD Thread)
     if (ptiCurrent->KeyboardLayout)
         UserDereferenceObject(ptiCurrent->KeyboardLayout);
 
+    if (ptiCurrent->spDefaultImc)
+    {
+        UserDereferenceObject(ptiCurrent->spDefaultImc);
```

Comment:
As object deref already happened, would it be possible/more_explicit/cleaner to close the (internal) handle instead?

Reply:
As can be seen from the (currently failing) tests, doing that extra dereference is unnecessary since the object is actually already destroyed.
```
*** Assertion failed: ObjHead->cLockObj < 0x10000
***   Source File: /win32ss/user/ntuser/object.c, line 615
```